config/metac.yaml

```diff
@@ -64,4 +64,51 @@ spec:
   hooks:
     sync:
       inline:
-        funcName: sync/command
\ No newline at end of file
+        funcName: sync/command
+---
+apiVersion: dope/v1
+kind: GenericController
+metadata:
+  name: finalize-command
+  namespace: dope
+spec:
+  watch:
+    apiVersion: dope.mayadata.io/v1
+    resource: commands
+  attachments:
+  # Delete pod
+  - apiVersion: v1
+    resource: pods
+    advancedSelector:
+      selectorTerms:
+      # select Pod if it has following label
+      - matchLabels:
+        matchReferenceExpressions:
+        - key: metadata.labels.job-name
+          operator: EqualsWatchName # match this lbl value against watch Name
+  # Delete job
+  - apiVersion: batch/v1
+    resource: jobs
+    advancedSelector:
+      selectorTerms:
+      # select job if its labels has following
+      - matchLabels:
+          command.dope.mayadata.io/controller: "true"
+        matchReferenceExpressions:
+        - key: metadata.labels.command\.dope\.mayadata\.io/name
+          operator: EqualsWatchName # match this lbl value against watch Name
+  - apiVersion: v1
+    resource: configmaps
+    advancedSelector:
+      selectorTerms:
+      # select ConfigMap if its labels has following
+      - matchLabels:
+          command.dope.mayadata.io/lock: "true"
+        matchReferenceExpressions:
+        - key: metadata.labels.command\.dope\.mayadata\.io/name
+          operator: EqualsWatchName # match this lbl value against watch Name
```

Comment:
These are my thoughts @mittachaitu 
- Job may have auto generated name
- Job will have Command's uid in its label
- Job will have Command's name in its label
- ConfigMap may have either auto generated name or named as cmd-<command's uid>
- ConfigMap will have Command's uid in its label
- ConfigMap will have Command's name in its label

The finalizer selector can make use of name & uid matches accordingly. I guess relying on name is not enough. It should be name & namespace. However, we might want to stick to UID for selection. Label key should be namespaced e.g. "command.dope.mayadata.io/name", "command.dope.mayadata.io/uid", etc.

Use of name selector might seem to be unnecessary since uid is sufficient. However, name will help to exclude the resource from the list quickly & hence improve the select performance. (_I hope the metac advanced selector matches by iterating the list of selectors. If name is the first selector then rejection will be fast._)

Reply:
@mittachaitu try to think deep on this!!!

Reply:
make use of uid to test changes